### DISCUSSION TITLE
`static` directory is too static 🗿

### DISCUSSION THREAD

**Discussion by mkreuzmayr** [resend/react-email#611] [Ideas]
### Goals

One of the following would be cool becuase I want my static files to work in `email dev` and production.

- Copy files from `static` and `public/static`, if they exist
- Detect NextJs in project root and search static in `public/static`
- Make the static directory path configurable

### Background

I am using `react-email` in a NextJs app and I want to serve the static folder with my app, this requires files to live in the `public` directory.

NextJs also shows a warning because the `static` directory is deprecated.
`The static directory has been deprecated in favor of the public directory. https://nextjs.org/docs/messages/static-dir-deprecated`

### Proposal

I would just copy files from `static` and `public/static`, whichever exists, it would be the simplest solution.

**Comment by bukinoshita** [comment]
A few questions

- What would be the best way to detect if user is using next?
- Does moving the files to your next app public folder, doesn't mean that the files will work in production.

In our case, we add this to the emails. But there's no way to know where the user is uploading their static files.

```ts
const baseUrl = process.env.VERCEL_URL
  ? `https://${process.env.VERCEL_URL}`
  : '';
```

**Comment by mkreuzmayr** [comment]
I understand that one does need to setup the paths correctly in a mail template to get assets working.

The easiest solution would be to make the static folder path configurable so people that use `next` can set it differently if they want. A cherry on top would be to check if `next.config.*` is present and use `public/static` by default. I think assuming that someone who starts using `next` in combination with `react-email` also stores assets in the `public` folder with it is viable.

It would be a great boost in DX if I just drag my assets to `public/static` and the mails work in dev and prod simultaneously without additional configuration.

Another thing is if one is using the static directory in the root, next.js shows a deprecation warning that could lead to confusion for some people.

**Comment by mkreuzmayr** [comment]
@bukinoshita I could set some time aside to implement a solution to this an file a PR.
